Firestore/Example/Tests/Util/FSTIntegrationTestCase.h

```diff
@@ -17,6 +17,8 @@
 #import <Foundation/Foundation.h>
 #import <XCTest/XCTest.h>
 
+#import <FirebaseCore/FIRAppInternal.h>
```

Comment:
For Objective-C headers, the convention is that we use forward declarations of Objective-C classes we refer to whose implementation we don't need. This reduces compilation time without any loss of safety.

The kinds of things we need actual declarations for (and the headers of that type in this file) are:
  * parent classes - `XCTest.h`
  * protocols we implement
  * categories we depend upon exporting - `XCTestCase+Await.h`

`Foundation.h` is included at the start of every header to make all the types of the Objective-C runtime available without along with macros that affect the classes we're declaration (e.g. `NS_ASSUME_NONNULL_BEGIN`).

This means you can declare methods that take and return `FIRApp *` just by declaring that the class exists with an `@class FIRApp;`. There are a bunch of examples below.

Note that this same guidance does not apply to C++, where types are substantially more complicated. For now, we still follow Google's internal "include what you use" rule for C++.